bears/general/SpaceConsistencyBear.py

```diff
@@ -1,22 +1,40 @@
-from coalib.bearlib import deprecate_settings
 from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
 from coalib.bears.LocalBear import LocalBear
 from coalib.results.Diff import Diff
 from coalib.results.Result import Result
+from coalib.bearlib.aspects import map_setting_to_aspect
+from coalib.bearlib.aspects.Formatting import (
+    Formatting,
+    TrailingSpace,
+    Indentation,
+    NewlineAtEOF
+)
 
 
-class SpaceConsistencyBear(LocalBear):
+class SpaceConsistencyBear(
+            LocalBear,
+            aspects={
+                'detect': [
+                    Indentation,
+                    NewlineAtEOF,
+                    TrailingSpace
+                ]},
+            languages=['Python']
+        ):
     LANGUAGES = {'All'}
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
     CAN_FIX = {'Formatting'}
 
-    @deprecate_settings(indent_size='tab_width')
+    @map_setting_to_aspect(
+        enforce_newline_at_EOF=NewlineAtEOF,
+        allow_trailing_whitespace=TrailingSpace,
```

Comment:
The actual value of a setting always depends on a taste and not on the pure existence of an aspect. So either the tastes have to be added above, or we need to somehow define implicit tastes for such single-taste aspects. So the actual problem here is that `map_setting_to_aspect` allows the above, but doesn't handle it properly...

cc @adhikasp

Reply:
@userzimmermann Thank you for all the very detailed feedback!
Am I right in saying this PR is now blocked until the above is implemented?

Reply:
@biscuitsnake Let's say so :) Do you want to implement those preconditions meanwhile?

Reply:
And you're welcome! Independently from your decision regarding the previous question ;)

Reply:
@userzimmermann All done I think 👍

Reply:
@biscuitsnake Hmm... Kind of ;)